vembrane/modules/annotate.py

```
@@ -0,0 +1,177 @@
+                tmp = tmp[3:]
+            for prefix in ["", "chr", "Chr", "CHR"]:
+                if prefix + tmp in available_chromsomes:
+                    chrom = prefix + tmp
+                    current_ann_data = chrom_ann_data[chrom]
+                    t = tree[chrom]
+
+        if chrom:
+            indices = np.fromiter((i for _, _, i in t[record.start]), dtype=np.int)
+            if len(indices):
+                env.update_data(current_ann_data[(np.array(indices))])
+                env.update_from_record(idx, record)
+                ann_values = env.table()
+
+                for v, expression_value in zip(
+                    map(lambda x: x["value"], config["annotation"]["values"]),
+                    ann_values,
+                ):
+                    if not v["number"] == ".":
+                        number = int(v["number"])
+                    else:
+                        number = -1
+
+                    if number == -1:
+                        expression_value = list(
+                            map(typeparser[v["type"]], expression_value)
+                        )
+                    elif number > 1:
+                        expression_value = list(
+                            map(typeparser[v["type"]], expression_value)
+                        )
+                        assert len(expression_value) == number
+                    else:
+                        # number == 1
+                        assert isinstance(expression_value, str) or not isinstance(
+                            expression_value, Iterable
+                        )
+                        expression_value = typeparser[v["type"]](expression_value)
+                    record.info[v["vcf_name"]] = expression_value
+
+        yield record
+
+
+def execute(args):
+    with open(args.config, "r") as stream:
+        try:
+            config = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
```

We should re-raise that error, no?

---

yes probably